packages/core-contracts/src/contracts/FleetDepositManager.sol

```diff
@@ -123,33 +123,29 @@ contract FleetDepositManager is ReentrancyGuard, ProtocolAccessManaged {
         if (fleetCommander == address(0)) revert InvalidParams();
         if (shareRecipient == address(0)) revert InvalidParams();
 
-        // Check if adapter is registered with BridgeRouter
-        if (!_isAdapterSupported(bridgeAdapter)) {
-            revert UnsupportedBridgeAdapter();
-        }
-
-        // Create fleet deposit compose message
-        bytes memory composeMessage = encodeFleetDepositMessage(
-            fleetCommander,
-            shareRecipient,
-            asset,
-            amount,
-            referralCode
-        );
-
-        // Transfer tokens from user to this contract, then approve adapter
+        // Transfer tokens from user to this contract, then approve BridgeRouter
         IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
-        IERC20(asset).forceApprove(bridgeAdapter, amount);
+        IERC20(asset).forceApprove(address(bridgeRouter), amount);
 
-        // Execute cross-chain deposit through the chosen adapter using standard interface
-        operationId = IFleetDepositAdapter(bridgeAdapter)
-            .sendFleetDepositToDestinationChain{value: msg.value}(
-            destinationChainId,
-            asset,
-            amount,
-            address(0), // destinationAdapter - not needed, adapter handles this
-            composeMessage,
-            adapterParams
+        // Create fleet deposit parameters
+        BridgeTypes.ExecuteUserFleetDepositParams memory params = BridgeTypes
+            .ExecuteUserFleetDepositParams({
+                destinationChainId: destinationChainId,
+                asset: asset,
+                amount: amount,
+                fleetCommander: fleetCommander,
+                shareRecipient: shareRecipient,
+                originalUser: msg.sender,
+                referralCode: referralCode,
+                options: BridgeTypes.BridgeOptions({
+                    specifiedAdapter: bridgeAdapter,
+                    adapterParams: adapterParams
+                })
+            });
```

Comment:
🛑it's not useing the `encodeFleetDepositMessage` so calling `executeUserFleetDeposit` in `BridgeRouter` won't work either - since there is no `message` (with the fleet deposit params) in the calldata